packages/astro/src/content/runtime.ts

```diff
@@ -315,6 +318,15 @@ type RenderResult = {
 	remarkPluginFrontmatter: Record<string, any>;
 };
 
+async function renderEntry(entry?: DataEntry) {
+	const Content = createComponent({
+		factory() {
+			return createHeadAndContent('', renderTemplate`${entry?.rendered?.html ?? ''}`);
+		},
+	});
+	return { Content };
+}
+
```

Comment:
I have no idea if this is a good way of doing this!

Reply:
Ha, seems like a minimal implementation to me! This would remove support for the `layout` frontmatter option in content collections, but that has been a maintenance burden for us. Happy to simplify for v2